apps/explorer/src/pages/checkpoints/CheckpointsTable.tsx

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useRpcClient } from '@mysten/core';
+import { ArrowRight12 } from '@mysten/icons';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo, useState } from 'react';
+
+import Pagination from '~/components/pagination/Pagination';
+import TabFooter from '~/components/tabs/TabFooter';
+import {
+    statusToLoadState,
+    type PaginationType,
+} from '~/components/transaction-card/RecentTxCard';
+import { TxTableCol } from '~/components/transaction-card/TxCardUtils';
+import { TxTimeType } from '~/components/tx-time/TxTimeType';
+import { CheckpointLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { PlaceholderTable } from '~/ui/PlaceholderTable';
+import { TableCard } from '~/ui/TableCard';
+import { Text } from '~/ui/Text';
+
+interface CheckpointsTableProps {
+    itemsPerPage?: number;
+    currentPage?: number;
+    paginationType?: PaginationType;
+    handlePageChange?: (page: number) => void;
+    shouldRefetch?: boolean;
+    refetchInterval: number;
+}
+
+export function CheckpointsTable({
+    itemsPerPage: initialItemsPerPage,
+    currentPage = 1,
+    handlePageChange,
+    paginationType,
+    shouldRefetch = false,
+    refetchInterval,
+}: CheckpointsTableProps) {
+    const rpc = useRpcClient();
+    const [itemsPerPage, setItemsPerPage] = useState(initialItemsPerPage || 20);
+    const countQuery = useQuery(
+        ['checkpoints', 'count'],
+        () => rpc.getLatestCheckpointSequenceNumber(),
+        { refetchInterval: shouldRefetch ? refetchInterval : false }
+    );
+
+    const { data } = useQuery(
+        ['checkpoints', { total: countQuery.data, itemsPerPage, currentPage }],
+        () =>
+            rpc.getCheckpoints({
+                limit: itemsPerPage,
+                descendingOrder: true,
+                cursor: countQuery.data! - (currentPage - 1) * itemsPerPage,
+            }),
+        { enabled: countQuery.isFetched, keepPreviousData: true }
+    );
+
+    const checkpointsTable = useMemo(
+        () =>
+            data
+                ? {
+                      data: data?.data.map((checkpoint) => ({
```

Comment:
super nit (sorry lol): not a huge fan of naming patterns like `data.data` when possible, perhaps you could rename the first `data` -> `checkpointsData`?